src/nvim/ex_getln.c

```diff
@@ -4444,11 +4467,12 @@ static int open_cmdwin(void)
 
   exmode_active = save_exmode;
 
-  // Safety check: The old window or buffer was deleted: It's a bug when
-  // this happens!
-  if (!win_valid(old_curwin) || !bufref_valid(&old_curbuf)) {
+  // Safety check: The old window or buffer was changed or deleted: It's a bug
+  // when this happens!
+  if (!win_valid(old_curwin) || !bufref_valid(&old_curbuf)
+      || old_curwin->w_buffer != old_curbuf.br_buf) {
     cmdwin_result = Ctrl_C;
-    emsg(_("E199: Active window or buffer deleted"));
+    emsg(_("E199: Active window or buffer changed or deleted"));
```

Comment:
I think this message should be changed to a `static const char []` variable above so that the variable name is included.